aten/src/ATen/test/backend_fallback_test.cpp

```diff
@@ -0,0 +1,141 @@
+  // Note: If op.callBoxed(stack) works for you, then that is preferrable because
+  // it's much faster and doesn't come with a dependency on JIT code.
+  // Instead, we take a path through the JIT operator registry, which has a boxed
+  // calling mechanism that works for all ops from native_functions.yaml.
+
+  auto s = Symbol::fromQualString(op.schema().name());
+  auto operators = torch::jit::getAllOperatorsFor(s);
+  // Find the exact match
+  std::shared_ptr<torch::jit::Operator> jit_op;
+  for (const auto& candidate_op : operators) {
+    auto candidate_schema = candidate_op->schema();
+    // NB: this is a VERY slow equality test
+    if (candidate_schema == op.schema()) {
+      jit_op = candidate_op;
+      break;
+    }
+  }
+  TORCH_INTERNAL_ASSERT(jit_op);
+
+  auto offset = jit_op->getOperation()(*stack);
+  TORCH_INTERNAL_ASSERT(offset == 0);
+}
+
+// Mode implementation
+
+void generic_mode_fallback(const c10::OperatorHandle& op, torch::jit::Stack* stack) {
+  override_call_count++;
+  c10::impl::ExcludeTensorTypeIdGuard guard(TensorTypeId::TESTING_ONLY_GenericModeTensorId);
+  callBoxedWorkaround(op, stack);
+}
+
+// Wrapper implementation
+
+struct GenericWrapperTensorImpl : public c10::TensorImpl {
+  explicit GenericWrapperTensorImpl(at::Tensor rep)
+    : TensorImpl(
+        c10::TensorTypeSet(c10::TensorTypeId::TESTING_ONLY_GenericWrapperTensorId),
+        rep.dtype(),
+        rep.device()
+        // TODO: propagate size!
+      )
+    , rep_(std::move(rep)) {}
+
+  at::Tensor rep_;
+};
+
+void generic_wrapper_fallback(const c10::OperatorHandle& op, torch::jit::Stack* stack) {
+  override_call_count++;
+
+  auto num_arguments = op.schema().arguments().size();
+  auto num_returns = op.schema().returns().size();
+
+  // Unwrap all arguments
+  auto args = torch::jit::pop(*stack, num_arguments);
+  for (size_t i = 0; i < num_arguments; i++) {
+    // TODO: Handle tensor list
+    if (args[i].isTensor()) {
+      auto* impl = args[i].unsafeToTensorImpl();
+      if (impl->type_set().has(TensorTypeId::TESTING_ONLY_GenericWrapperTensorId)) {
+        auto* wrapper = static_cast<GenericWrapperTensorImpl*>(impl);
+        torch::jit::push(*stack, wrapper->rep_);  // no move!
+      } else {
+        torch::jit::push(*stack, std::move(args[i]));
+      }
+    } else {
+      torch::jit::push(*stack, std::move(args[i]));
+    }
+  }
+
+  callBoxedWorkaround(op, stack);
+
+  // Rewrap outputs
+  auto rets = torch::jit::pop(*stack, num_returns);
+  for (size_t i = 0; i < num_returns; i++) {
+    // TODO: Handle tensor list
+    if (args[i].isTensor()) {
+      torch::jit::push(*stack, at::detail::make_tensor<GenericWrapperTensorImpl>(std::move(args[i]).toTensor()));  // yes move!
+    } else {
+      torch::jit::push(*stack, std::move(args[i]));
+    }
+  }
+}
+
+struct Environment {
+  c10::RegistrationHandleRAII registry1 = c10::Dispatcher::singleton().registerBackendFallbackKernel(
```

Comment:
Afaict all your tests for backend fallback kernels register them by a direct call into `c10::Dispatcher::singleton().registerBackendFallbackKernel()`, as opposed to through `torch::RegisterOperators` that's used for other types of kernels (including catchAllKernels).  Do you plan to expose the backend fallback registration through `torch::RegisterOperators` as well, or will registering directly onto the dispatcher instance continue to be the plan?  It's usable for me either way.

Reply:
this is still under discussion. For now, please use the dispatcher instance directly.